r/R/dplyr-funcs.R

```diff
@@ -58,15 +58,27 @@ NULL
 #' @keywords internal
 #'
 register_binding <- function(fun_name, fun, registry = nse_funcs) {
-  name <- gsub("^.*?::", "", fun_name)
-  namespace <- gsub("::.*$", "", fun_name)
+  qualified_name <- fun_name
+  unqualified_name <- gsub("^.*?::", "", qualified_name)
```

Comment:
Don't need global substitution, and use `:+` to remove both `::` and `:::`

```suggestion
  unqualified_name <- sub("^.*?:+", "", fun_name)
```

Reply:
the greedy matching with `:+` doesn't seem to be working (I get your point, I will look into it a bit):
``` r
fun_name <- "somePkg::some_fun"

gsub("^.*?::", "", fun_name)
#> [1] "some_fun"
sub("^.*?:+", "", fun_name)
#> [1] ":some_fun"
```

<sup>Created on 2022-07-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
it looks like `+` needs to be surrounded by curly brackets 🙃 
``` r
sub("^.*?:{+}", "", "somePkg::some_fun")
#> [1] "some_fun"
sub("^.*?:{+}", "", "somePkg:::some_fun")
#> [1] "some_fun"
```

<sup>Created on 2022-07-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
Done.